src/core/ext/filters/client_channel/resolver/dns/event_engine/service_config_helper.cc

```diff
@@ -0,0 +1,132 @@
+// Copyright 2023 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <grpc/support/port_platform.h>
+
+#include "src/core/ext/filters/client_channel/resolver/dns/event_engine/service_config_helper.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <algorithm>
+#include <map>
+#include <utility>
+#include <vector>
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+
+#include "src/core/lib/gprpp/debug_location.h"
+#include "src/core/lib/gprpp/status_helper.h"
+#include "src/core/lib/iomgr/gethostname.h"
+#include "src/core/lib/json/json.h"
+
+namespace grpc_event_engine {
+namespace experimental {
+
+namespace {
+using ::grpc_core::Json;
+
+bool ValueInJsonArray(const Json::Array& array, const char* value) {
+  for (const Json& entry : array) {
+    if (entry.type() == Json::Type::STRING && entry.string_value() == value) {
+      return true;
+    }
+  }
+  return false;
+}
+
+}  // namespace
+
+absl::StatusOr<std::string> ChooseServiceConfig(
```

Comment:
What refactoring would be required here?  It seems like it's basically a drop-in replacement.  The only real difference is that it returns a `StatusOr<>` instead of having a `grpc_error_handle` output parameter.

Reply:
This looks great!